rustup-init.sh

```diff
@@ -400,30 +402,50 @@ downloader() {
         get_ciphersuites_for_curl
         _ciphersuites="$RETVAL"
         if [ -n "$_ciphersuites" ]; then
-            curl --proto '=https' --tlsv1.2 --ciphers "$_ciphersuites" --silent --show-error --fail --location "$1" --output "$2"
+            _err=$(curl --proto '=https' --tlsv1.2 --ciphers "$_ciphersuites" --silent --show-error --fail --location "$1" --output "$2" 2>&1)
+            _status=$?
         else
             echo "Warning: Not enforcing strong cipher suites for TLS, this is potentially less secure"
             if ! check_help_for "$3" curl --proto --tlsv1.2; then
                 echo "Warning: Not enforcing TLS v1.2, this is potentially less secure"
-                curl --silent --show-error --fail --location "$1" --output "$2"
+                _err=$(curl --silent --show-error --fail --location "$1" --output "$2" 2>&1)
+                _status=$?
             else
-                curl --proto '=https' --tlsv1.2 --silent --show-error --fail --location "$1" --output "$2"
+                _err=$(curl --proto '=https' --tlsv1.2 --silent --show-error --fail --location "$1" --output "$2" 2>&1)
+                _status=$?
             fi
         fi
+        if [ -n "$_err" ]; then
+            echo "$_err" >&2
+            if echo "$_err" | grep -q 404$; then
+                err "unsupported platform $3"
+            fi
+        fi
+        return $_status
     elif [ "$_dld" = wget ]; then
         get_ciphersuites_for_wget
         _ciphersuites="$RETVAL"
         if [ -n "$_ciphersuites" ]; then
-            wget --https-only --secure-protocol=TLSv1_2 --ciphers "$_ciphersuites" "$1" -O "$2"
+            _err=$(wget --https-only --secure-protocol=TLSv1_2 --ciphers "$_ciphersuites" "$1" -O "$2" 2>&1)
+            _status=$?
         else
             echo "Warning: Not enforcing strong cipher suites for TLS, this is potentially less secure"
             if ! check_help_for "$3" wget --https-only --secure-protocol; then
                 echo "Warning: Not enforcing TLS v1.2, this is potentially less secure"
-                wget "$1" -O "$2"
+                _err=$(wget "$1" -O "$2" 2>&1)
+                _status=$?
             else
-                wget --https-only --secure-protocol=TLSv1_2 "$1" -O "$2"
+                _err=$(wget --https-only --secure-protocol=TLSv1_2 "$1" -O "$2" 2>&1)
+                _status=$?
+            fi
+        fi
+        if [ -n "$_err" ]; then
+            echo "$_err" >&2
+            if echo "$_err" | grep -q 'ERROR 404: Not Found\.$'; then
```

Comment:
Will `ERROR 404: Not Found.` me the same in non-english locales?

Reply:
No, nice findings. I forgot that different locale may have different messages.
```
--2020-07-27 16:02:23--  https://static.rust-lang.org/rustup/dist/armv7-unknown-linux-musleabihf/rustup-init
已载入 CA 证书“/etc/ssl/certs/ca-certificates.crt”
正在解析主机 static.rust-lang.org (static.rust-lang.org)... 54.192.150.81, 54.192.150.109, 54.192.150.55, ...
正在连接 static.rust-lang.org (static.rust-lang.org)|54.192.150.81|:443... 已连接。
已发出 HTTP 请求，正在等待回应... 404 Not Found
2020-07-27 16:02:23 错误 404：Not Found。
```

```suggestion
            if echo "$_err" | grep -q ' 404: Not Found\.$'; then
```

Reply:
Oh wait, the period is different. I guess we need to check the second last line `404 Not Found`.

Reply:
I worry that the `Not Found` text happens to be what AWS returns too.  Is there a reason you didn't just grep for `404` like you do in the cURL cases?

Reply:
Because I bet AWS won't change that.

Reply:
Fair enough